### DISCUSSION TITLE
Translations in Navigation bar

### DISCUSSION THREAD

**Discussion by Hadjimina** [amannn/next-intl#126] [Q&A]
Hi all,

I have been playing around with next-intl for a bit now and implemented almost everything. The one area I can't get it to work is in the navigation bar that I define in the `_app.tsx` file s.t. it appears on every page and I don't have to worry about it. 
Any idea how I can make the translation work in the CustomNavbar component? 
I can't use the approach for a normal page with `getStaticProps` since that only works on pages not in `_app.tsx`. 
Ideally, I would not need to wrap every page individually in the CustomNavbar component.

Excerpt from `_app.tsx`:
```
    <NextIntlProvider messages={pageProps.messages}>
      <Head>
      ...
      </Head>

     
        <CustomNavbar scrolleRef={viewport}>
          <main className="main">
              <Component {...pageProps} />
          </main>
        </CustomNavbar>
     
    </NextIntlProvider>
```

Cheers

**Comment by amannn** [comment]
Hi @Hadjimina,

what kind of difficulties are you facing with providing the messages for the `CustomNavbar`? Can you include those translations from `getStaticProps` of individual pages? I.e. a page would fetch its own messages, but also the ones that are considered shared—in this case the `CustomNavbar`.

**Reply by Hadjimina** [reply]
Oh wow, for some reason I thought I had to include these messages in the same file where I put the CustomNavbar in the return statement. Nvm, super easy fix. Thanks!
